File: manprobackend/app/Http/Controllers/DonationGoalController.php
<?php

namespace App\Http\Controllers;

use App\Models\DonationGoal;
use Illuminate\Http\Request;
use Illuminate\Support\Facades\Log;

class DonationGoalController extends Controller
{
    public function index()
    {
        try {
            $goals = DonationGoal::with('donationType')
                ->get()
                ->map(function ($goal) {
                    return [
                        'id' => $goal->id,
                        'type' => $goal->donationType ? $goal->donationType->name : 'Unknown',
                        'target_quantity' => (int) $goal->target_quantity,
                        'current_quantity' => (int) $goal->current_quantity,
                        'percentage' => (int) $goal->percentage
                    ];
                });

            return response()->json([
                'status' => 'success',
                'data' => $goals
            ]);
        } catch (\Exception $e) {
            Log::error('Error fetching donation goals: ' . $e->getMessage());
            
            return response()->json([
                'status' => 'error',
                'message' => 'Failed to fetch donation goals',
                'debug_message' => config('app.debug') ? $e->getMessage() : null
            ], 500);
        }
    }
} 

File: manproapp/lib/features/bagian_utama/controllers/donation_goal_controller.dart
import 'package:get/get.dart';
import 'package:http/http.dart' as http;
import 'dart:convert';
import '../Models/donation_goal.dart';
import '../../../utils/constants/api_constants.dart';

class DonationGoalController extends GetxController {
  RxList<DonationGoal> goals = <DonationGoal>[].obs;
  RxBool isLoading = false.obs;
  RxString error = ''.obs;

  @override
  void onInit() {
    super.onInit();
    fetchDonationGoals();
  }

  Future<void> fetchDonationGoals() async {
    try {
      isLoading.value = true;
      error.value = '';

      final response = await http.get(Uri.parse('${url}donation-goals'));
      
      if (response.statusCode == 200) {
        final decodedResponse = json.decode(response.body);
        print('API Response: ${response.body}'); // Debug print
        
        if (decodedResponse['status'] == 'success' && decodedResponse['data'] != null) {
          final List<dynamic> goalsData = decodedResponse['data'];
          goals.value = goalsData.map((goal) => DonationGoal.fromJson(goal)).toList();
        } else {
          error.value = decodedResponse['message'] ?? 'Failed to load donation goals';
        }
      } else {
        error.value = 'Server error: ${response.statusCode}';
      }
    } catch (e) {
      print('Error fetching goals: $e'); // Debug print
      error.value = 'Error: ${e.toString()}';
    } finally {
      isLoading.value = false;
    }
  }
} 